src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1082,25 +1082,30 @@ private void parseGroupLabel(GroupDef g, Element e) {
         Vector<String> usedAtts = new Vector<>();
         usedAtts.addElement(REF_ATTR);
 
-        String ref = e.getAttributeValue("", REF_ATTR);
+        String labelItextId = getItextReference(e);
+        g.setTextID(labelItextId);

```

Comment:
do we want to `setTextID` even if `labelItextID` is `null`?

Reply:
yeah as that will just assign it to `null`, so would not matter at all.